### PR TITLE
joe: bump to upstream v 4.2

### PR DESCRIPTION
**Pull request by aTanW**
Signed-off-by: Vitaly Protsko <villy@sft.ru>
---
 Makefile    |    4
 files/joerc |  494 ++++++++++++++++++++++++++++++++++++++----------------------
 2 files changed, 318 insertions(+), 180 deletions(-)

### CODE REVIEW

File: utils/joe/files/joerc [hunk 1]
```diff
@@ -7,46 +6,55 @@
...
 -guess_non_utf8
+-guess_utf16
 -guess_crlf
 -guess_indent
 -menu_above
 -transpose
--joe_state
--restore
--mouse
+ -joe_state
```
Comment (hnyman) [inline utils/joe/files/joerc]: This line ``` + -joe_state``` looks suspiciousyl like an accidental whitespace problem. Or is the leading space intentional?

File: utils/joe/files/joerc [hunk 2]
```diff
@@ -261,6 +328,7 @@ hnext		^[ .
 hprev		^[ ,
 math		^[ m
 math		^[ M
+ math		^[ ^M
```
Comment (hnyman) [inline utils/joe/files/joerc]: Same also here. several of the additions seem to contain a leading space. Looks strange.

**Comment by aTanW** [comment]
All that spaces are ok.
This is comment char in that syntax.
-joe_state commented out to save flash cycles
all others are setups for different kinds of terminals for people who wants precisely tune all commands

File: utils/joe/files/joerc (hunk 1 above)
Comment (aTanW) [inline utils/joe/files/joerc]: Pls, look at comment below

File: utils/joe/files/joerc (hunk 2 above)
Comment (aTanW) [inline utils/joe/files/joerc]: Pls, look at comment below

**Pull request merged**